hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/keygen/TestTimestampBasedKeyGenerator.java

```diff
@@ -378,4 +382,46 @@ public void test_ExpectsMatch_MultipleInputFormats_ShortDate_OutputCustomDate()
     baseRow = genericRecordToRow(baseRecord);
     assertEquals("04/01/2020", keyGen.getPartitionPath(baseRow));
   }
+
+  @Test
+  public void test_ExpectsMatch_LogicalType_Date() throws IOException {
+    LocalDate today = LocalDate.now();
+    baseRecord.put("dob", (int) today.toEpochDay());
+    properties = this.getBaseKeyConfig(
+            "DATE",
+            "yyyy-MM-dd'T'HH:mm:ssZ,yyyy-MM-dd'T'HH:mm:ss.SSSZ,yyyyMMdd",
+            "",
+            null,
+            "yyyy-MM-dd",
+            DateTimeZone.getDefault().getID());
+
+    properties.setProperty(KeyGeneratorOptions.PARTITIONPATH_FIELD_OPT_KEY, "dob");
+    BuiltinKeyGenerator keyGen = new TimestampBasedKeyGenerator(properties);
+    HoodieKey hk1 = keyGen.getKey(baseRecord);
+    Assertions.assertEquals(today.toString(), hk1.getPartitionPath());
+
+    baseRow = genericRecordToRow(baseRecord);
+    assertEquals(today.toString(), keyGen.getPartitionPath(baseRow));
+  }
+
+  @Test
+  public void test_ExpectsMatch_LogicalType_Timestamp() throws IOException {
```

Comment:
I see we have added LocalDate, LocalDatetime, Date and Timestamp support to source code, but tests are added only for 2 of these. can we please add for rest as well.

Reply:
The tests check for all 4. When GenericRecord is being used to get the partitionPath, then date field goes as LocalDate, whereas when it gets converted to Row, then it is represented as Date. Similarly, for timestamp, when GenericRecord is used, the field is represented as LocalDateTime and when converted to Row, it is represented as Timestamp.

Reply:
thanks for clarifying